### DISCUSSION TITLE
VCards

### DISCUSSION THREAD

**Discussion by TitoBahe** [devlikeapro/waha#641] [Q&A] [answered]
Does the VCard sent to me by others appear in the message.any payload?

I am not receiving it, neither when I send the VCard nor when someone else sends it to me.

Here is an example of the payload I receive when I send the Vcard, the actual Vcard is not metioned here:
```
{
  "event": "message.any",
  "session": "xkCvZLdM9TSwxl5k36hU-clientId0",
  "metadata": {
    "user.id": "xkCvZLdM9TSwxl5k36hU-clientId0",
    "user.email": "email@example.com"
  },
  "me": { 
    "id": "558185464605@c.us", 
    "pushName": "TitoTest" 
  },
  "payload": {
    "id": "true_558199777011@c.us_D1BED4582A7DF252025AFA2C1A345951",
    "timestamp": 1731772343,
    "from": "558199777011@c.us",
    "fromMe": true,
    "hasMedia": false,
    "ack": 1,
    "ackName": "SERVER",
    "replyTo": { 
      "id": "", 
      "participant": "", 
      "body": null 
    },
    "_data": {
      "key": { ... },
      "messageTimestamp": 1731772343,
      "pushName": "TitoTest",
      "broadcast": false,
      "status": 2,
      "message": { ... }
    }
  },
  "engine": "NOWEB",
  "environment": {
    "version": "2024.10.14",
    "engine": "NOWEB",
    "tier": "PLUS",
    "browser": "/usr/bin/chromium"
  }
}
```

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment, marked as answer]
it's inside `_data.message.contactMessage.vcard` right now

Probably we could parse it and add in `contacts` array :thinking: feel free to create a feature request if you need that!
https://github.com/devlikeapro/waha/issues/new/choose

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
